packages/netlify-cms-core/src/formats/__tests__/yaml.spec.js

```diff
@@ -13,5 +62,73 @@ describe('yaml', () => {
     test('outputs valid yaml', () => {
       expect(yaml.toFile([])).toEqual('[]\n');
     });
+
+    test('should sort keys', () => {
+      expect(yaml.toFile({ a: 'a', b: 'b', c: 'c', d: 'd' })).toEqual('a: a\nb: b\nc: c\nd: d\n');
+      expect(yaml.toFile({ a: 'a', b: 'b', c: 'c', d: 'd' }, ['d', 'b', 'a', 'c'])).toEqual(
+        'd: d\nb: b\na: a\nc: c\n',
+      );
+      expect(yaml.toFile({ a: 'a', b: 'b', c: 'c', d: 'd' }, ['d', 'b', 'c'])).toEqual(
+        'a: a\nd: d\nb: b\nc: c\n',
+      );
+    });
+
+    test('should add comments', () => {
+      expect(
+        yaml.toFile({ a: 'a', b: { c: { d: 'd' } } }, [], { a: 'Comment a', 'b.c': 'Comment c' }),
+      ).toEqual('# Comment a\na: a\nb:\n  # Comment c\n  c:\n    d: d\n');
+
+      const expected = `# template comment
+template: post
+# title comment
+title: title
+# image comment
+image: /media/netlify.png
+# date comment
+date: "2020-04-02T13:27:48.617Z"
```

Comment:
Only difference I could find between `js-yaml` is that `yaml` uses double quotes and not single quotes (e.g. `'2020-04-02T13:27:48.617Z'`.
Tried https://eemeli.org/yaml/#tag-stringifier-options but that quotes the keys as well.